modules/navier_stokes/include/userobjects/RhieChowMassFlux.h

```diff
@@ -0,0 +1,226 @@
+namespace libMesh
+{
+class Elem;
+class MeshBase;
+}
+
+/**
+ * User object responsible for determining the face fluxes using the Rhie-Chow interpolation in a
+ * segregated solver that uses the linear FV formulation.
+ */
+class RhieChowMassFlux : public GeneralUserObject,
+                         public BlockRestrictable,
+                         public NonADFunctorInterface,
+                         public FaceArgInterface
+{
+public:
+  static InputParameters validParams();
+  RhieChowMassFlux(const InputParameters & params);
+
+  /// Get the face velocity (used in advection terms)
+  Real getMassFlux(const FaceInfo & fi) const;
+
+  /// Initialize the container for face velocities
+  void initFaceMassFlux();
+  /// Update the values of the face velocities in the containers
+  void computeFaceMassFlux();
+  /// Update the cell values of the velocity variables
+  void computeCellVelocity();
+
+  virtual void meshChanged() override;
+  virtual void initialize() override;
+  virtual void execute() override {}
+  virtual void finalize() override {}
+  virtual void initialSetup() override;
+
+  /**
+   * Update the momentum system-related information
+   * @param momentum_systems Pointers to the momentum systems which are solved for the momentum
+   * vector components
+   * @param pressure_system Reference to the pressure system
+   * @param momentum_system_numbers The numbers of these systems
+   */
+  void linkMomentumPressureSystems(std::vector<LinearSystem *> momentum_systems,
+                                   const LinearSystem & pressure_system,
+                                   const std::vector<unsigned int> & momentum_system_numbers);
+
+  /**
+   * Computes the inverse of the digaonal (1/A) of the system matrix plus the H/A components for the
+   * pressure equation plus Rhie-Chow interpolation.
+   */
+  void computeHbyA(bool verbose);
+
+  bool hasFaceSide(const FaceInfo & fi, const bool fi_elem_side) const override;
+
+protected:
+  void setupCellVolumes();
+
+  /// Populate the face values of the H/A and 1/A fields
+  void
+  populateCouplingFunctors(const std::vector<std::unique_ptr<NumericVector<Number>>> & raw_hbya,
+                           const std::vector<std::unique_ptr<NumericVector<Number>>> & raw_Ainv);
+
+  /**
+   * Check the block consistency between the passed in \p var and us
+   */
+  template <typename VarType>
+  void checkBlocks(const VarType & var) const;
+
+  /// The \p MooseMesh that this user object operates on
+  MooseMesh & _moose_mesh;
+
+  /// The \p libMesh mesh that this object acts on
+  const libMesh::MeshBase & _mesh;
+
+  /// The dimension of the mesh, e.g. 3 for hexes and tets, 2 for quads and tris
+  const unsigned int _dim;
+
+  /// The thread 0 copy of the pressure variable
+  MooseLinearVariableFVReal * const _p;
+
+  /// The thread 0 copy of the x-velocity variable
+  std::vector<MooseLinearVariableFVReal *> _vel;
+
+  LinearFVAnisotropicDiffusion * _p_diffusion_kernel;
```

Comment:
```suggestion
  /// Pointer to the pressure diffusion kernel in the pressure correction equation
  LinearFVAnisotropicDiffusion * _p_diffusion_kernel;
```
I need to check those equations to be sure. I feel like a tech editor making up acronyms

Reply:
Great suggestion, thanks Guillaume!